src/storage-types/src/sources/mysql.rs

```diff
@@ -63,6 +64,37 @@ impl RustType<ProtoMySqlColumnRef> for MySqlColumnRef {
     }
 }
 
+pub struct IntoMySqlColumnRefError(String);
+
+impl fmt::Display for IntoMySqlColumnRefError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "{}", self.0)
+    }
+}
+
+impl TryFrom<UnresolvedItemName> for MySqlColumnRef {
+    type Error = IntoMySqlColumnRefError;
+
+    /// Convert an UnresolvedItemName into a MySqlColumnRef
+    fn try_from(item: UnresolvedItemName) -> Result<Self, Self::Error> {
+        let mut iter = item.0.into_iter();
+        match (iter.next(), iter.next(), iter.next()) {
+            (Some(schema_name), Some(table_name), Some(column_name)) => Ok(MySqlColumnRef {
+                // We need to be careful not to accidentally introduce postgres/mz-style
+                // quoting in the schema_name, table_name, or column_name fields, which
+                // happens if naively using the to_string()/fmt method on the Ident values.
+                schema_name: schema_name.into_string(),
+                table_name: table_name.into_string(),
+                column_name: column_name.into_string(),
```

Comment:
wild bug, omg

Reply:
tell me about it haha. It took me a while for my brain to notice `into_string` and `to_string` were different 😅